src/audio/player.js

```diff
@@ -316,1 +1,316 @@
+            shuffled[randomIndex] = temporaryValue;
+        }
+
+        const unpauseEmbed = new Discord.MessageEmbed()
+            .setTitle('🔀 重新排序音樂')
+            .setColor(colors.success);
+        this._channel.send({
+            embeds: [unpauseEmbed],
+        }).catch(this.noop);
+        this.songs = shuffled;
+    }
+
+    async playStream() {
+        if (!this._songs[0]?.rawData.full) {
+            try {
+                this._songs[0].rawData = await play.video_info(this._songs[0].url);
+                this._songs[0].rawData.full = true;
+            } catch (e) {
+                log.error(e.message);
+                const errorEmbed = new Discord.MessageEmbed()
+                    .setTitle('🙁 載入音樂時發生錯誤')
+                    .setDescription(
+                        '載入音樂時發生了一點小錯誤...\n' +
+            '錯誤內容:\n' +
+            '```\n' + e.message + '\n```')
+                    .setColor(colors.danger);
+                this._channel.send({
+                    embeds: [errorEmbed],
+                }).catch(this.noop);
+                return;
+            }
+        }
+
+        let stream;
+        try {
+            stream = await play.stream(this._songs[0].url);
+        } catch (e) {
+            log.error(e.message);
+            const errorEmbed = new Discord.MessageEmbed()
+                .setTitle('🙁 載入音樂時發生錯誤')
+                .setDescription(
+                    '載入音樂時發生了一點小錯誤...\n' +
+          '錯誤內容:\n' +
+          '```\n' + e.message + '\n```')
+                .setColor(colors.danger);
+            this._channel.send({
+                embeds: [errorEmbed],
+            }).catch(this.noop);
+            return;
+        }
+        this._audio = createAudioResource(stream.stream, {
+            inputType: stream.type,
+            metadata: this._songs[0],
+        });
+        this._player.play(this._audio);
+    }
+
+    get ping() {
+        return this._connection.ping;
+    }
+
+    get nowplaying() {
+        return this._audio.metadata;
+    }
+
+    get playTime() {
+        return this._audio.playbackDuration / 1000;
+    }
+
+    get songs() {
+        return this._songs;
+    }
+
+    handelIdle() {
+        this._noticeMessage?.delete().catch(this.noop);
+
+        this._songs.shift();
+        if (this._songs.length <= 0) {
+            const endEmbed = new Discord.MessageEmbed()
+                .setTitle('👌 序列裡的歌曲播放完畢');
+            this._channel.send({
+                embeds: [endEmbed],
+            }).catch(this.noop);
+        } else {
+            this.playStream();
+        }
+    }
+
+    async handelPlaying() {
+        const playingEmbed = new Discord.MessageEmbed()
+            .setTitle(`🎵 目前正在播放 ${this._audio.metadata.title}`)
+            .setURL(this._audio.metadata.url)
+            .setThumbnail(this._audio.metadata.thumbnail)
+            .setColor(colors.success);
+        this._noticeMessage = await this._channel.send({

```

Comment:
Codacy has a fix for the issue: [Expected indentation of 8 spaces but found 4.](https://app.codacy.com/gh/blackcatbot/blackcat/pullRequest?prid=8779465)
```suggestion
        this._noticeMessage = await this._channel.send({
```